### DISCUSSION TITLE
Rough walls

### DISCUSSION THREAD

**Discussion by czc-zju** [AMReX-Combustion/PeleLMeX#231] [Q&A]
In PeleLM/PeleLMex， can the walls in the figure be modeled and meshed? Or can this wall be embedded?
This wall is built based on  the expression:  f=cos(2pai x)*cos(2pai z), pai =3.14.
![1a72b8f9b83a95b7dee309682159715](https://github.com/AMReX-Combustion/PeleLMeX/assets/81547457/92819a62-74da-4573-91ab-b2fb2174ec78)

**Comment by baperry2** [comment]
**Background:** Embedded boundary geometry definition in PeleLMeX comes from AMReX. For very simple geometries like spheres or planes, the geometry may be specified directly in the input file using the `eb2` prefix (see the `EB_BackwardStepFlame` tutorial for an example). You can look at the AMReX source code `amrex/Src/EB/AMReX_EB2.cpp` for the full list of shapes and corresponding input file parameters that can be defined in this way. More complex shapes may be defined by specifying an `EBUserDefined` function in a problem-specific `EBUserDefined.H` (see the `ChallengeProblem` Case for an example). In this function, you can take various shapes defined in AMReX and combine them and/or apply different transformations. See the [AMReX documentation](https://amrex-codes.github.io/amrex/docs_html/EB.html#) for more details.

**Answer:** I do not believe that sinusoids are currently possible by default with either approach for specifying the EB geometry described above. However, I don't think it would be too difficult for you to add a class to give this capability, taking [amrex/Src/EB/AMReX_IF_Plane.H](https://github.com/AMReX-Codes/amrex/blob/development/Src/EB/AMReX_EB2_IF_Plane.H) as an example. You basically just need to define a function of (x,y,z) that is positive in the solid, negative within the fluid, and zero on the surface, which would be pretty easy for your geometry. If you do that, there shouldn't be any issues with simulating the rough channel case, provided you use enough resolution to capture the scale of the roughness.

**Comment by czc-zju** [comment]
Thank you very much for your suggestions. I have made some attempts in the past few days, but I encountered some problems. Whenever I try to use PeleLMex for calculations, I always get an error. I'm not sure what went wrong. Here are the configuration files for PeleLMex and the EB_plane file:
PeleLMex Settings File:
[pelelmex.zip](https://github.com/AMReX-Combustion/PeleLMeX/files/12326257/pelelmex.zip)
AMReX_EB2_IF_Parser.H:
![image](https://github.com/AMReX-Combustion/PeleLMeX/assets/81547457/f88efb8b-ee3d-4dc3-97ba-d867972920cc)

I have also tried using peleC, and it works fine. However, the velocity at the embedded body surface should be zero, but it's not in my case. I suspect there might still be some issues. Here are the results:
![image](https://github.com/AMReX-Combustion/PeleLMeX/assets/81547457/33c04ce8-4686-46aa-8e33-479c1e1cbc18)
![image](https://github.com/AMReX-Combustion/PeleLMeX/assets/81547457/e34eeacc-570f-433f-affd-c6eaff174831)

**Comment by baperry2** [comment]
Can you give some more details on the error you are seeing when you try this in pelelmex?

**Comment by czc-zju** [comment]
![image](https://github.com/AMReX-Combustion/PeleLMeX/assets/81547457/83a4045a-2743-494c-9a5f-a292d012b08c)

**Comment by baperry2** [comment]
You are likely using insufficient resolution to adequately resolve the roughness features. There should not be multiple EB surfaces cutting through any grid cell. Double check the function you are using to specify the surface and make sure the wavelength of the roughness elements is much larger than your grid spacing. I'd recommend that you start out by using a very large wavelength and low amplitude. I wouldn't be surprised if adding a slight phase shift could also be useful in avoiding this error.

You could also experiment with the `eb2.cover_multiple_cuts` input file flag from AMReX, though I'm not familiar at all with what that functionality does or if it could help.

I'm going to move this conversation to a discussion because it's not about existing capability in PeleLMeX.

**Reply by czc-zju** [reply]
Thank you very much for your help!  You are right. When I used sufficient resolution to adequately resolve the roughness features,  it works well.
